syntax = "proto3";

message RequestEnvelop {
  int32 unknown1 = 1; // Always 2
  int64 rpc_id = 3; // Random
  repeated Request requests = 4; // Future requests?
  Unknown6 unknown6 = 6; // Let's hope it's not a must
  fixed64 gps_x = 7;
  fixed64 gps_y = 8;
  fixed64 gps_z = 9;
  AuthInfo auth = 10; // Used only in first request
  TokenData token = 11; // A token received after first request. Doesn't change in session.
  int64 time_delta = 12; // Time between RPCs?

  message Request {
    int32 type = 1;
    bytes message = 2;
  }

  message Unknown6 {
    int32 unknown1 = 1;
    Unknown2 unknown2 = 2;

    message Unknown2 {
      bytes unknown1 = 1;
    }

  }
  message AuthInfo {
    string provider = 1; // Google
    JWT token = 2; // Token received

    message JWT {
      string contents = 1;
    }
  }

  message TokenData {
    bytes token = 1;
    int64 timestamp = 2;
    bytes sig = 3; // Possibly a SHA1 signature
  }

  message GetMapObjectsRequest{
    //repeated uint64 cell_id = 1;
    bytes cell_id = 1;
    //repeated int64 since_time_ms = 2;
    bytes since_time_ms = 2;
    fixed64 player_lat = 3;
    fixed64 player_lng = 4;
  }
}
